test/url-to-options.js

```diff
@@ -0,0 +1,56 @@
+import url from 'url';
+import test from 'ava';
+import urlToOptions from '../source/url-to-options';
+
+const githubURL = 'https://user:password@github.com:443/';
+
+test('converts node legacy URL to options', t => {
+	const parsedURL = url.parse(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: null,
+		hostname: 'github.com',
+		href: githubURL,
+		path: '/null',
+		pathname: '/',
+		port: 443,
+		protocol: 'https:',
+		search: null
+	};
+
+	t.deepEqual(options, expected);
+});
+
+test('converts URL to options', t => {
+	const parsedURL = new url.URL(githubURL);
```

Comment:
Keep in mind that since Node.JS 10 URL is a global :)
I think we should duplicate [that TODO comment](https://github.com/sindresorhus/got/blob/master/source/normalize-arguments.js#L2) and paste it there:
```// TODO: Use the `URL` global when targeting Node.js 10```

Reply:
To be honest, I don't feel these are useful at all. I'd simply say that if we end up preferring global over imported `URL` and we're at a point where our lowest target is Node 10, anybody can feel free to change it.

Until then, even assuming it is a change we don't want to forget to make, I don't see the point in having the comment.

Regardless, I don't like being too opinionated on a project I've been away from especially on a trivial two line change 😊 , so fixed!